GUIDELINES.md

```diff
@@ -0,0 +1,95 @@
+   like ones.
+3. Consider possible objections against the proposal and research them
+   as part of DIP document.
+4. Stick to formal and technical language in general, avoid colloquialisms. A
+   DIP should be formulated like a scientific paper, as opposed to an op-ed
+   article.
+5. Provide sufficient techniqal details that can be used as a specification
+   by D compiler developers.
+
+Be prepared for a lot of work. There are always many ideas proposed but
+many fewer developers commited to pursuing the idea to the final stages of
+evaluation. The DIP system is _not_ for submitting undeveloped ideas, it is
+a process for formal approval of language changes.
+
+Rest of the guidelines provides more detailed explanation of requirements.
+
+## Justification
+
+Any DIP must focus on "why" at least as much as "what" and often even more.
+At this stage of the D language development changes need to justify their
+existence with great value. Doubt about the necessity or value of a
+gchange will be a strong motivator for rejection.
+
+It is important to research any alternative approaches to solving the same
+problem and explain your choice. If there are any relevant success or failure
+stories in other programming languages, consider referring to them with
+explanation of how they might apply to D.
+
+Redundancy of existing features should not be used as precedent and
+justification for adding another redundant feature. In a similar way, existing
+language flaw cannot be used as an excuse for adding more functionality with
+a similar flaw.
+
+It is important to evaluate costs of adding something the language. Something
+that uses existing syntax and builds upon established semantics is considered
+of much less weight than a proposal requiring completely new syntax or
+concept.
+
+## Examples
+
+The motivation section should showcase examples that currently are not
+possible or are contorted in D, followed by their alternative implementation
+using the proposed feature. At best, the example would be taken from existing
+code (e.g. found in dub, phobos, etc). Next best is code that defines a
+plausible artifact.
+
+Keep in mind that without such examples to back it, any reasoning in favor of
+proposed feature may be completely discarded. This is also why examples from
+existing code are preferred - D is a practical language and even theoretically
+sound proposal is not important enough if it doesn't offer any substantial
+improvements to code that actually gets written.
+
+When possible, provide same example snippet implemented with both existing D
+language status quo and using proposed functionality to show the improvement in
+most obvious way.
+
+## Technical details
+
+Anyone attemtping to propose a language change is expected to be sufficiently
+knowledgeable in existing semantics that are affected by the proposal, as
+well as compiler internals, to provide formal explanation of required changes.
```

Comment:
Let's loosen this requirement a bit:

s/compiler internals/compiler technology/